### ISSUE TITLE
Do not force UTF-8 encoding on frozen response bodies

### ISSUE THREAD

**Issue by rochlefebvre** [elastic/elastic-transport-ruby#63]
The excellent [webmock](https://github.com/bblimke/webmock) http stubbing library has recently started [freezing the response body](https://github.com/bblimke/webmock/blob/master/CHANGELOG.md#3190) stub. Tests that make use of webmock to mock Elasticsearch responses now encounter a `FrozenError: can't modify frozen String` here:

https://github.com/elastic/elastic-transport-ruby/blob/0e52364ebc810d567f86b6837714a0b6c4bcebc9/lib/elastic/transport/transport/response.rb#L32

Changing a frozen string's encoding causes this error. After testing the following change locally, the error goes away:

```ruby
@body = body.force_encoding('UTF-8') if body.respond_to?(:force_encoding) && !body.frozen?
```

If UTF-8 encoding is a must, then frozen strings need to be duplicated beforehand. This was [rejected](https://github.com/elastic/elasticsearch-ruby/issues/725) a few [times](https://github.com/elastic/elasticsearch-ruby/issues/726) before 🤷 .

**Comment by picandocodigo** [comment]
Thank you for reporting this @rochlefebvre! I just released elastic-transport [v8.1.2](https://github.com/elastic/elastic-transport-ruby/releases/tag/v8.1.2) and [v8.2.3](https://github.com/elastic/elastic-transport-ruby/releases/tag/v8.2.3) which address this issue.

**Comment by rochlefebvre** [comment]
> Thank you for reporting this @rochlefebvre! I just released elastic-transport [v8.1.2](https://github.com/elastic/elastic-transport-ruby/releases/tag/v8.1.2) and [v8.2.3](https://github.com/elastic/elastic-transport-ruby/releases/tag/v8.2.3) which address this issue.

Thank you very much for addressing this problem! I have confirmed that the release fixes several broken CI pipelines in our org.
